package/Editor/Images/Images.cs

```diff
@@ -141,12 +141,24 @@ public void DeleteImage(string _id)
 
         public static ImageDataStorage.ImageData GetImageDataById(string _id)
         {
-            return imageDataList.Find(x => x.Id == _id);
+            var imageData = imageDataList.Find(x => x.Id == _id);
+            if (imageData == null)
+                imageData = DataCache.instance.GetImageDataById(_id); //try to get from Datacache (if it has just been prompted)
+            if (imageData == null)
```

Comment:
you can return imageData at this stage already, no need to check again if it is null to return null